src/main/kotlin/org/rust/cargo/project/CargoProjectDescription.kt

```diff
@@ -27,8 +27,8 @@ class CargoProjectDescription private constructor(
     ) {
         val isModule: Boolean get() = source == null
         val libTarget: Target? get() = targets.find { it.isLib }
-
         val contentRoot: VirtualFile? get() = VirtualFileManager.getInstance().findFileByUrl(contentRootUrl)
+        val isExternal: Boolean get() = targets.asSequence().filter { it.crateRoot != null }.any { it.isLib }
```

Comment:
The idea here is to produce regular hyperlinks for the source code we control (it's in our workspace) from the source code we don't (external dependencies). It doesn't depend on the particular target we're currently running. We produce regular hyperlinks for our own code and grayed ones for everything else. This is how this feature works in Java.

`isExternal` seems to work pretty reliably now for the described task, while `isModule` returns `true` for both our own crates and external dependencies.

But maybe things are more complicated than I think and I don't understand your point? We could look at a particular example that you think won't work.